packages/api/cms-api/src/dam/files/files.utils.ts

```diff
@@ -66,3 +67,46 @@ export const calculatePartialRanges = (size: number, range: string): { start: nu
         contentLength: end - start + 1,
     };
 };
+
+const recursivelyFindJSInSvg = (svg: SvgRootNode | SvgNode): boolean => {
+    if (svg.type === "text") {
+        // is TextNode -> can't contain JS
+        return false;
+    }
+
+    if (svg.type === "element") {
+        // is ElementNode -> can be a script tag or have an event handler attribute
+
+        if (svg.tagName && svg.tagName.toLowerCase() === "script") {
+            // is script tag
+            return true;
+        }
+
+        if (svg.properties && Object.keys(svg.properties).some((propKey) => propKey.toLowerCase().startsWith("on"))) {
+            // has event handler
+            return true;
+        }
+    }
+
+    // is ElementNode or RootNode -> can contain JS in its children
+    if (svg.children.length > 0) {
+        let containsJS = false;
+
+        for (const child of svg.children) {
+            if (typeof child === "object") {
+                if (recursivelyFindJSInSvg(child)) {
+                    containsJS = true;
```

Comment:
You may use `return true` here to stop checking early. If there's a single instance of JS code, we can directly abort the upload